src/pages/campaigns/NewCampaign.tsx

```diff
@@ -0,0 +1,250 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as z from "zod";
+import { format } from "date-fns";
+import { Type, FileText, Send, Calendar as CalendarIcon } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { useEffect, useState } from "react";
+import MDXComponent from "@/components/NewsletterComponents/MDXComponent";
+import Sidebar from "@/components/Sidebar";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Calendar } from "@/components/ui/calendar";
+import { cn } from "@/lib/utils";
+import { useRecoilState } from "recoil";
+import { markdownState } from "@/atoms/atoms";
+import { marked } from "marked";
+const formSchema = z.object({
+  title: z
+    .string()
+    .max(120, { message: "Title must be 120 characters or less." })
+    .min(1, { message: "Title is required." }),
+  blurb: z
+    .string()
+    .max(120, { message: "Blurb must be 120 characters or less." })
+    .min(1, { message: "Blurb is required." }),
+  content: z.string(),
+  launchDate: z.date({ required_error: "Launch date is required." }),
```

Comment:
Make sure to sync in with the backend on the date-type. Dates can be send as strings or as a separate date because JSON supports both.

Reply:
Backend its DateTime.
Date validation is also added.